### ISSUE TITLE
Postgres supports in-place enum value renames these days

### ISSUE THREAD

**Issue by aseemk** [prisma/prisma#23569]
## Problem

I renamed an enum value in my Prisma schema:

```prisma
enum ExampleEnum {
  // before: FOO
  // after:
  BAR

  @@map("example_enum")
}
```

and Prisma generated the following unsafe migration:

```sql
/*

  Warnings:

  - The values [FOO] on the enum `example_enum` will be removed. If these variants are still used in the database, this will fail.

*/
-- AlterEnum
BEGIN;
CREATE TYPE "example_enum_new" AS ENUM ('BAR');
ALTER TABLE "example_table" ALTER COLUMN "example_column" TYPE "example_enum_new" USING ("type"::text::"example_enum_new");
ALTER TYPE "example_enum" RENAME TO "example_enum_old";
ALTER TYPE "example_enum_new" RENAME TO "example_enum";
DROP TYPE "example_enum_old";
COMMIT;
```

As the warning states, this just won't work if there are any rows with the old enum value. So not a terribly helpful migration.

## Suggested solution

Postgres these days makes this way easier, and supports existing data:

```sql
ALTER TYPE example_enum RENAME VALUE 'FOO' TO 'BAR';
```

Looks like this was added in Postgres 10, so over 6 years ago now:

https://www.postgresql.org/docs/10/sql-altertype.html

## Alternatives

None considered.

## Additional context

No add'l context.

**Comment by SevInf** [comment]
How would migrate know that you want to rename `FOO` to `BAR` in this case, rather than remove `FOO` and add `BAR` variant at the same time?

**Comment by aseemk** [comment]
Yeah great q. Perhaps the CLI could detect this possibility and ask if that was your intent. (It already detects the current dropped data possibility and asks you to confirm.)

**Comment by eminel9311** [comment]
i'm trying change data of enum from "CUSTOMER" to "CUSTOMERs" but when i run migrate dev --create-only , this migrate file generated
![image](https://github.com/prisma/prisma/assets/32560843/65b38551-2f68-43f4-a2b3-84b5ace38483)
Then i run migrate dev and have notice
![image](https://github.com/prisma/prisma/assets/32560843/d4fe0a6f-2211-443f-9c6c-bc8ba27e3b0e)
Please support me.Thanks

**Comment by janpio** [comment]
Please open a bug report and provide a full reproduction of what you are seeing @eminel9311 - thank you. Your problem is unrelated to what this _feature request_ is about.
